### PR TITLE
feat(ui): share transaction history modal

### PR DESCRIPTION
**Pull request by peps**
This PR adds the UI for sharing specific transactions in the History list. 

To enable: 

1) in `useShareRewardStore.ts` change `sharingEnabled: false` to `sharingEnabled: true`
2) Make sure you're logged in to airdrop ( it's disabled unless you're logged in )

Note: This can be merged in it's disabled state, but DO NOT launch it publicly until the share card work is completed on the airdrop website side. 

CleanShot 2024-10-30 at 17 45 52

CleanShot 2024-10-30 at 17 46 22

Other changes in this PR:

1. Added a blur to the GreenModal component background cover
2. Added a darkened cover to the sidebar when the history is open ( to match the design )

### CODE REVIEW

File: src/store/useShareRewardStore.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    item: Transaction | null;
+}
+
+interface Actions {
+    setShowModal: (showModal: boolean) => void;
+    setItemData: (item: Transaction | null) => void;
+}
+
+const initialState: State = {
+    sharingEnabled: false,
```
Comment (peps) [inline src/store/useShareRewardStore.ts:16]: Set this to `true` to enable the ui

**Review by brianp — CHANGES_REQUESTED** [review]
When "Show history" is clicked the UX completely falls apart, and the app breaks. Let me know if I'm doing something wrong. I've enabled the feature as described, and everything looks normal until that point.

**Comment by peps** [comment]
> When "Show history" is clicked the UX completely falls apart, and the app breaks. Let me know if I'm doing something wrong. I've enabled the feature as described, and everything looks normal until that point.

Looks like a bad merge, fixing it now

**Comment by peps** [comment]
@brianp should be good now

**Comment by brianp** [comment]
PR now works after merge fixes. You may want to apply similar modal sizing fixes though. It is scrollable, so I'm going to merge. Up to you to submit sizing fixes if that sorta thing is up your alley.

image
image

**Review by brianp — APPROVED** [review]

**Comment by Anziskavds** [comment]
Tested unsuccessfully on MacBook Air 13 inch M1 2020 on Testnet v0.5.63.

image

**Pull request merged**